source/use-cases/sensitive-data-encryption.txt

```diff
@@ -0,0 +1,312 @@
+- deterministic or random. We can also specify the BSON Type of the field we are encrypting. 
+In addition, we can specify the key we want to use. This is particularly important, because
+it means that we can use different keys for different fields.
+
+In the following example, we will go over creation of a schema for a collection which represents patient data
+in a medical office. Some fields use deterministic encryption, and others use random (probabilistic encryption). 
+
+The structure for our schema will be as follows:
+
+.. code::
+
+    fullName: String
+    address: Encrypted String using random encryption
+    ssn: Encrypted Int using deterministic encryption
+    last4SSN: Int
+    patientInfo: Object
+    patientInfo.phone Encrypted Int using deterministic encryption
+    patientInfo.provider String
+    medicalRecords: Encrypted array of documents using random encryption
+
+Explanation of encryption choices:
+``````````````````````````````````
+* address is randomly encrypted because it is more secure for strings to be randomly encrypted to prevent hackers from decrypting by building a cipher-dictionary
+* ssn is deterministically encrypted because deterministic encryption provides filtering ability 
+* patientInfo.phone is deterministically encrypted for the same reason as ssn, and also it provides an example of encryption with a nested document
+* medicalRecords are randomly encrypted because entire array's are only encryptable with random encryption
+
+
+.. tabs::
+
+   tabs:
+
+     - id: java-jsonschema-generation
+       name: "Java"
+       content: |
+
+         The following code creates a collection using the schema above and inserts it into a database using MongoDB.
+         For the "Define a JSON Schema" section we've abstracted the createSecureClient, to focus on creation of a schema.
+         But that function uses getJSONSchema and passes it a dataKey id in order to create a client which uses the JSONSchema
+         we want to use. 
+
+         .. code-block:: java
+
+            import com.mongodb.*;
+            import com.mongodb.client.MongoClient;
+            import com.mongodb.client.MongoClients;
+            import com.mongodb.client.MongoCollection;
+            import com.mongodb.client.model.vault.DataKeyOptions;
+            import com.mongodb.client.vault.ClientEncryptions;
+
+            import org.bson.BsonDocument;
+
+            import java.util.Base64;
+            import java.util.Map;
+
+            import org.json.*;
+
+            public class FieldLevelEncryptionDemo {
+                static String dbName = "fieldLevelEncryptionDemo001";
+                static String collectionName = "patients";
+                public static void main(String[] args) {
+                    MongoCollection secureCollection = createSecureClient().getDatabase(dbName).getCollection(collectionName);
+                    ...
```

Comment:
On line 99, you mentioned:

>  The following code creates a collection using the schema above and inserts it into a database using MongoDB.

I think we should make sure that the description and code match.

Reply:
How about: "The following code creates a JSONSchema object that enforces encryption on various fields"

Reply:
The statement you're proposing is fine, but what I was thinking that we should show in this section is:

- Create the JSONSchema object
- Add it to a collection (or create the collection with it specified, if that is easier) using `runCommand`. Explain why you add the validator to the collection
- Mention that the JSONSchema object will be used in the AutoEncryptionSettings in the client setup (upcoming section). Explain why you would add it to the autoencryption settings.

Does that work for you?

Reply:
Those three points works for me to add to the doc. Would you suggest putting those after that above line ^ ("The following code creates a collection using the schema above and inserts it into a database using MongoDB.")